core/src/main/java/org/jdbi/v3/core/extension/ExtensionHandler.java

```diff
@@ -52,25 +52,26 @@ public interface ExtensionHandler {
      * @param args Optional arguments for the handler
      * @return The return value for the method that was bound to the extension handler. Can be null
      * @throws Exception Any exception from the underlying code
+     * @deprecated call {@link #createInvoker(Object)} instead
      */
-    Object invoke(HandleSupplier handleSupplier, Object target, Object... args) throws Exception;
+    @Deprecated(forRemoval = true)
+    default Object invoke(HandleSupplier handleSupplier, Object target, Object... args) throws Exception {
+        return createInvoker(target).invoke(handleSupplier, args);
+    }
 
     /**
-     * Called after the method handler is constructed to pre-initialize any important
-     * configuration data structures.
-     *
-     * @param config the method configuration to use for warming up
+     * Bind this extension handler to a target instance and return an Invoker.
+     * @param target The target object on which the handler should operate
      */
-    @Beta
-    default void warm(ConfigRegistry config) {}
+    ExtensionHandler.Invoker createInvoker(Object target);
```

Comment:
here you use the qualified name.

Reply:
given the comment below, wouldn't "bindExtensionHandler" or similar match more closely what this code does?